src/Controller/Admin/RolesController.php

```diff
@@ -41,4 +47,18 @@ class RolesController extends AdministrationBaseController
         'name' => 'string',
         'description' => 'text',
     ];
+
+    /**
+     * Save data
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function save(): ?Response
+    {
+        $response = parent::save();
+        $roles = Hash::combine((array)$this->apiClient->get('/roles'), 'data.{n}.id', 'data.{n}.attributes.name');
+        Cache::write(self::CACHE_KEY_ROLES, $roles);
```

Comment:
Since this cache is already being created elsewhere when needed, I'd just purge it in this action, so that we don't duplicate the logic: in case we ever need to change the cached value format, we have to change it both in `PermissionsTrait` and here.

To avoid this scenario, I'd just clear it here, and it will be written by PermsTrait the first time it's needed again.

```suggestion
        Cache::delete(self::CACHE_KEY_ROLES);
```

Reply:
Cache invalidation should be performed in roles removal as well. So I added the `remove` method with a `Cache::delete(self::CACHE_KEY_ROLES)`.